docs/in-depth.md

```diff
@@ -0,0 +1,225 @@
+A good way to understanding the sortition pool design is to understand the
+original problem and naive approach, and then understand the optimizations away
+from that naive solution.
+
+### Original Problem and Naive Approach
+
+Say that we have a list of operators like `[Alice, Bob, Carol, David]`, and
+each operator has a different amount of stake in the pool `{Alice: 30, Bob: 20:
+Carol: 40: David: 50}`. We need to select a group of operators (say, 3), where
+each operator selected has a chance to be selected equal to their *portion* of
+the total amount of stake in the pool. It's okay to select one operator
+multiple times. We might end up with a group that looks like `[David,
+Carol, David]`.
+
+In our example, the chance that we select Alice is `30 / (30 + 20 + 40 + 50) =
+0.214`.
+
+In order to select a single *slot* in our group, we might write something simple like:
+
+```
+function chooseOperator(operators) {
+  const operatorNames = Object.keys(operators)
+
+  let totalWeight = 0
+  for (let i = 0; i < operatorNames.length; i++) {
+    const operatorName = operatorNames[i]
+    totalWeight += operators[operatorName]
+  }
+
+  let randomValue = Math.floor(Math.random() * totalWeight)
+
+  let chosenOperator = undefined
+  for (let i = 0; i < operatorNames.length; i++) {
+    const operatorName = operatorNames[i]
+    const operatorStake = operators[operatorName]
+    if (operatorStake > randomValue) {
+      chosenOperator = operatorName
+      break;
+    } else {
+      randomValue -= operatorStake
+    }
+  }
+  return chosenOperator
+}
+chooseOperator({"Alice": 30, "Bob": 20, "Carol": 40, "David": 50})
+```
+
+In English, we figure out the total weight of the pool, and then generate a
+random number in `[0, totalWeight)`. Then, we assign a sort of cumulative
+distribution to each of the operators: Alice is picked if that random number is
+in `[0, 30)`, Bob is picked if the random number is in `[20, 50)`, Carol is
+picked if the random number is in `[50, 90)`, and David is picked if the random
+number is in `[90, 140)`.
+
+Say our random number was 64. We know that's in Carol's range, but how do we
+determine that? First, we check if 64 > Alice's 30, and it isn't, so we
```

Comment:
This should be `First, we check if 64 < Alice's 30, and it isn't, so we subtract...`

Reply:
https://github.com/keep-network/sortition-pools/pull/157/commits/2b14a6fcb99e0934de72015513470389f99e857e